spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkInputPartition.java

```diff
@@ -66,6 +66,10 @@ public <T extends ScanTask> ScanTaskGroup<T> taskGroup() {
     return (ScanTaskGroup<T>) taskGroup;
   }
 
+  public <T extends ScanTask> boolean allTasksOfType(Class<T> javaClass) {
```

Comment:
Curious question here, do we assume a single task type for every taskGroup or not?  

Just wondering if we can optimize it to just check the type of taskGroup.

Reply:
The idea is to be able to check whether all tasks conform to a particular known parent type. For instance, for changelog tasks we only check if all are children of `ChangelogScanTask`. Then the reader itself may downcast it to a particular child type (e.g. `ChangelogRowReader`).

We can't check the type of taskGroup due to Java type erasure.

Reply:
Do we allow a taskgroup to have tasks of different types? I didn't think so. But I don't think it costs us that much to just check all of them instead of just the first element.

Reply:
Makes sense.  Yea my question was Russell's question, if there's one to one relation always (defined by TaskGroup).  But I guess there's nothing in TaskGroup preventing different tasks that are subclass of <T>, although that would be strange.

Reply:
That's true, `ScanTaskGroup` may have arbitrary tasks.